pkg/catalogv2/system/system.go

```diff
@@ -399,41 +406,47 @@ func (m *Manager) isInstalled(namespace, name, version, minVersion string, desir
 			return false, "", nil, err
 		}
 
-		desiredValue = map[string]interface{}{}
-		if err := json.Unmarshal(patchedJSON, &desiredValue); err != nil {
+		current, err := semver.NewVersion(r.Chart.Metadata.Version)
+		if err != nil {
 			return false, "", nil, err
 		}
 
-		ver, err := semver.NewVersion(release.Chart.Metadata.Version)
+		desired, err := semver.NewVersion(latestVersion)
 		if err != nil {
 			return false, "", nil, err
 		}
 
+		if minVersion == "" {
+			// Use exact version.
+			if !current.Equal(desired) {
+				return false, desired.String(), desiredValues, nil
+			}
+		}
+
 		if minVersion != "" {
 			min, err := semver.NewVersion(minVersion)
 			if err != nil {
 				return false, "", nil, err
 			}
 			if desired.LessThan(min) {
-				logrus.Errorf("available chart version (%s) for %s is less than the min version (%s) ", desired, name, min)
+				logrus.Errorf("available chart version (%s) for %s is less than the min version (%s) ", desired, r.Chart.Name(), min)
 				return false, "", nil, repo.ErrNoChartName
 			}
-			if min.LessThan(ver) || min.Equal(ver) {
+			if min.LessThan(current) || min.Equal(current) {
 				// If the current deployed version is greater or equal than the min version but configuration has changed, return false and upgrade with the current version
 				if !bytes.Equal(patchedJSON, actualValueJSON) {
-					return false, release.Chart.Metadata.Version, desiredValue, nil
+					return false, r.Chart.Metadata.Version, desiredValues, nil
 				}
-				logrus.Debugf("Skipping installing/upgrading desired version %v for release %v, current version %v is greater or equal to minimal required version %v", desired.String(), name, ver.String(), minVersion)
+				logrus.Debugf("Skipping installing/upgrading desired version %v for release %v, current version %v is greater or equal to minimal required version %v", desired.String(), r.Chart.Name(), current.String(), minVersion)
```

Comment:
Nit:

You are using `chart.Name()` here, but the prior version was using `name` which seems to be the release name (the log message also indicates that this is supposed to be the release name as well).